ibis/backends/duckdb/__init__.py

```diff
@@ -40,6 +38,38 @@
     import torch
 
 
+class _SettingsPassThrough:
+    def __init__(self, con):
+        self.con = con
+
+    def __getitem__(self, key):
+        with self.con.begin() as con:
+            if value := con.exec_driver_sql(
+                f"select value from duckdb_settings() where name = '{key}'"
+            ).cursor.fetchone():
+                return value[0]
+            else:
+                raise KeyError(key)
+
+    def __setitem__(self, key, value):
+        with self.con.begin() as con:
+            con.exec_driver_sql(f"SET {key}='{value}'")
+
+    def __repr__(self):
+        with self.con.begin() as con:
+            table = con.exec_driver_sql(
+                "select name, value from duckdb_settings()"
+            ).cursor.fetch_arrow_table()
+
+        rows = ["{"]
+        rows.extend(
+            f"'{key}': '{value}',"
+            for key, value in zip(table["name"].to_pylist(), table["value"].to_pylist())
+        )
+        rows.append("}")
+        return "\n".join(rows)
```

Comment:
Perhaps we can simplify this a bit:

```suggestion
        with self.con.begin() as con:
            kv = con.exec_driver_sql(
                "select map(array_agg(name), array_agg(value)) from duckdb_settings()"
            ).scalar()

        return repr(dict(zip(kv["key"], kv["value"])))
```